lib/IntegrationConfiguration.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const yaml = require('js-yaml');
+const fs = require('fs');
+
+let integrationPath;
+let privateKeyPath;
+
+/**
+     * Constructor for Asset Compute integration configuration
+     *
+     * @param {string} [integrationFile=] Configuration file path for integration (optional)
+     *                                    If no file path is passed in, ENV will be checked
+     * @param {string} [privateKeyFile=] Configuration file path for private key file when supplying JSON (optional)
+     *                                    If no file path is passed in, ENV will be checked
+     */
+async function getConfiguration(integrationFile, privateKeyFile) {
+
+    privateKeyPath = privateKeyFile || process.env.ASSET_COMPUTE_PRIVATE_KEY_FILE_PATH;
+    integrationPath = integrationFile || process.env.ASSET_COMPUTE_INTEGRATION_FILE_PATH;
+
+    return validate();
+}
+
+function validate() {
+
+    if (!fs.existsSync(integrationPath)) {
```

Comment:
by replacing `existsSync` with another call (`stat` or `lstat` if you stay with only node's `fs`. There are more options if you decide to use `fs-extra`, like `pathExists`).